btree/btree.c

```diff
@@ -2812,8 +2982,16 @@ static bool ff_isoverflow(const struct nd *node, int max_ksize,
 	    ff_crctype_get(node) == M0_BCT_BTREE_ENC_RAW_HASH)
 		crc_size = sizeof(uint64_t);
 
-	return (ff_space(node) < h->ff_ksize + ff_valsize(node) + crc_size) ?
-	       true : false;
+	if (IS_EMBEDDED_INDIRECT(node)) {
+		if (h->ff_used == 0 || (h->ff_used < h->ff_max_recs &&
+		    (ff_space(node) >= h->ff_ksize + ff_valsize(node) +
+		    		       crc_size)))
```

Comment:
The check here should be :
if (he->ff_used == 0 || h->ff_used < h_ff_max_recs)
There does not seem to be a need for checking if ff_space is giving enough space as we are having a fixed format.

Reply:
Done.